Update release notes for TensorFlow 2.3.0

tensorflow-jenkins opened the pull request:
This PR is intentionally incomplete. One of the Release Owners for 2.3.0
needs to fill in the internal release notes for this version before the PR gets
submitted. Click on the :pencil2: icon in the header for `RELEASE.md` under
"Files Changed" above.

tensorflow-jenkins committed:
Insert release notes place-fill

goldiegadde committed:
Update RELEASE.md

goldiegadde committed:
Update RELEASE.md

goldiegadde committed:
Update RELEASE.md

angerson committed:
Added point about tf.sysconfig.get_build_info()

goldiegadde committed:
Update RELEASE.md

martinwicke commented on RELEASE.md:
which bug?

goldiegadde replied:
Updated.

martinwicke commented on RELEASE.md:
what does this do?

goldiegadde replied:
this is a change to a test case, confirmed with Yanhua author of the change and removing this.

martinwicke commented on RELEASE.md:
Is there a link we could add?

goldiegadde replied:
Updated.

goldiegadde commented on RELEASE.md:
```diff
@@ -1,15 +1,199 @@
+   breaking change only impacts `tf.image.extract_glimpse` and
+   `tf.compat.v2.image.extract_glimpse` API endpoints. The behavior of
+   `tf.compat.v1.image.extract_glimpse` does not change. The behavior of
+   exsiting C++ kernel `ExtractGlimpse` does not change as well, so saved
```
Done!

goldiegadde commented on RELEASE.md:
```diff
@@ -1,15 +1,199 @@
+   `tf.compat.v2.image.extract_glimpse` API endpoints. The behavior of
+   `tf.compat.v1.image.extract_glimpse` does not change. The behavior of
+   exsiting C++ kernel `ExtractGlimpse` does not change as well, so saved
+   models will not be impacted.
```
Done

goldiegadde commented on RELEASE.md:
```diff
@@ -1,15 +1,199 @@
+    * Add Bessel functions of order 0,1 to `tf.math.special`.
+    * `tf.divide` now always returns a tensor to be consistent with documentation and other APIs.
+  * `tf.image`:
+    * Replaces [`tf.image.non_max_suppression_padded`](https://www.tensorflow.org/api_docs/python/tf/image/non_max_suppression_padded?hl=en&version=nightly) with a new implementation that supports batched inputs, which is considerably faster on TPUs and GPUs. Boxes with area=0 will be neglected. Existing usage with single inputs should still work as before.
```
Done

goldiegadde commented on RELEASE.md:
```diff
@@ -1,15 +1,199 @@
+    * Add Bessel functions of order 0,1 to `tf.math.special`.
+    * `tf.divide` now always returns a tensor to be consistent with documentation and other APIs.
+  * `tf.image`:
+    * Replaces [`tf.image.non_max_suppression_padded`](https://www.tensorflow.org/api_docs/python/tf/image/non_max_suppression_padded?hl=en&version=nightly) with a new implementation that supports batched inputs, which is considerably faster on TPUs and GPUs. Boxes with area=0 will be neglected. Existing usage with single inputs should still work as before.
```
I can update to rc0 , but 2.3 API will only be updated after final.

goldiegadde replied:
never mind, I updated all the api docs links from nightly to 2.3 version.

goldiegadde committed:
Update RELEASE.md

goldiegadde commented on RELEASE.md:
```diff
@@ -1,15 +1,199 @@
+    * `@tf.function` from SavedModel no longer ignores args after a `RaggedTensor` when selecting the concrete function to run.
+    * Fix save model issue for ops with a list of functions.
+    * Add `tf.saved_model.LoadOptions` with [`experimental_io_device`](https://www.tensorflow.org/api_docs/python/tf/saved_model/LoadOptions) as arg with default value `None` to choose the I/O device for loading models and weights.
+     * Update `tf.saved_model.SaveOptions` with [`experimental_io_device`](https://www.tensorflow.org/api_docs/python/tf/saved_model/SaveOptions?version=nightly) as arg with default value `None` to choose the I/O device for saving models and weights.
```
Updated to use 2.3 version

goldiegadde commented on RELEASE.md:
@guptapriya can you help address the above comment

guptapriya replied:
Yeah we should remove the part about "since there was no way to confirm,.. " i don't think that is relevant to the reader here. Sadly the document on this is not public. 

How about rephrasing as:

Allow `var.assign` on `MirroredVariables` with `aggregation=NONE` in replica context. Previously this would raise an error. We now allow this because it many users and library writers find using `.assign` in replica context to be more convenient, instead of having to use `Strategy.extended.update` which was the previous way of updating variables in this situation.

goldiegadde replied:
Added the above changes.

sun51 commented on RELEASE.md:
```diff
@@ -1,15 +1,199 @@
+
+### TF Core:
+  * Set `tf2_behavior` to 1 to enable V2 for early loading cases.
+  * Add a function to dynamically choose the implementation based on underlying device placement.
```
Add execute_fn_for_device function to dynamically choose the implementation based on underlying device placement.

goldiegadde replied:
Done.

caisq committed:
Add mentioned Debugger V2 to r2.3 release notes

caisq committed:
Grammar tweaks in the Debugger V2 bullet point

goldiegadde commented on RELEASE.md:
```diff
@@ -1,15 +1,199 @@
+     * Update `tf.saved_model.SaveOptions` with [`experimental_io_device`](https://www.tensorflow.org/api_docs/python/tf/saved_model/SaveOptions?version=nightly) as arg with default value `None` to choose the I/O device for saving models and weights.
+  * GPU
+    * No longer includes PTX kernels for GPU except for sm_70 to reduce binary size.
+  * Profiler
```
added as a separate section.

goldiegadde commented on RELEASE.md:
no issue ids :(

goldiegadde replied:
action item noted, resolving this for now.

goldiegadde committed:
Update RELEASE.md

goldiegadde commented on RELEASE.md:
removing this, as it is a minor fix after confirming with Yanhua, author of the change.

goldiegadde committed:
Update RELEASE.md

goldiegadde commented on RELEASE.md:
Updated. PTAL

goldiegadde commented on RELEASE.md:
```diff
@@ -1,15 +1,199 @@
+
+  * [`tf.distribute.TPUStrategy`](https://www.tensorflow.org/api_docs/python/tf/distribute/TPUStrategy) is now a stable API and no longer considered experimental for TensorFlow. (earlier `tf.distribute.experimental.TPUStrategy`).
+
+  * TF Profiler introduces two new tools: a memory profiler to visualize your model’s memory usage over time and a python tracer which allows you to trace python function calls in your model. Usability improvements include better diagnostic messages and profile options to customize the host and device trace verbosity level.
```
Done.

angerson committed:
Reword tf.sysconfig.get_build_info notice

An internal bug report revealed that "support" could be mistaken for "official support," which is not intended.

lamberta commented on RELEASE.md:
```diff
@@ -1,15 +1,199 @@
+    * Add `tf.saved_model.LoadOptions` with [`experimental_io_device`](https://www.tensorflow.org/api_docs/python/tf/saved_model/LoadOptions) as arg with default value `None` to choose the I/O device for loading models and weights.
+     * Update `tf.saved_model.SaveOptions` with [`experimental_io_device`](https://www.tensorflow.org/api_docs/python/tf/saved_model/SaveOptions?version=nightly) as arg with default value `None` to choose the I/O device for saving models and weights.
+  * GPU
+    * No longer includes PTX kernels for GPU except for sm_70 to reduce binary size.
```
Can perhaps use some of this copy from the updated /install/gpu guide:

On systems with NVIDIA® Ampere GPUs (CUDA architecture 8.0) or newer, kernels are JIT-compiled from PTX and TensorFlow can take over 30 minutes to start up. This overhead can be limited to the first start up by increasing the default JIT cache size with: '`export CUDA_CACHE_MAXSIZE=2147483648`'.

goldiegadde replied:
Updated.

goldiegadde committed:
Update RELEASE.md
